### PR TITLE
[MLLIB] SPARK-4362: Added classProbabilities method for Naive Bayes

### PR DESCRIPTION
**Pull request by actgardner**
Added methods which accept an RDD or array and return a map of (label -> posterior prob.) for each input set rather than only returning the key with the maximum value.

### CODE REVIEW

**Comment by srowen** [comment]
@alanctgardner have you had a look at @jkbradley 's feedback? I'm wondering this is still live. It needs a rebase if so.

File: mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala [hunk 1]
```diff
@@ -65,6 +66,25 @@ class NaiveBayesModel private[mllib] (
   override def predict(testData: Vector): Double = {
     labels(brzArgmax(brzPi + brzTheta * testData.toBreeze))
   }
+
+  def classProbabilities(testData: RDD[Vector]):
```
Comment (alanctgardner) [inline mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala]: Sorry for the delay, I have no strong preference but "predictProbabilities" makes sense for consistency. I can make that change and the style ones mentioned.

My stats background is not super-strong, @jatinpreet seemed to imply there's a correctness issue with this PR. Can anyone comment on if I've got the math wrong?

**Comment by jkbradley** [comment]
@alanctgardner  That will be great if you change it to predictProbabilities; thanks.  I agree with what @jatinpreet was saying about the correctness, and with @srowen 's comment on how to fix it: The value of ```brzPi + brzTheta * testData.toBreeze``` is a log probability, which needs to be exponentiated before you normalize it here: [https://github.com/apache/spark/pull/3626/files?diff=split#diff-6d8eff78be2fb624d4a076db334208a4R84]

Could you please rebase off of master and make these couple of updates?  After that, I can make a final pass.  Thanks!

**Comment by srowen** [comment]
Mind closing this PR? if it's not going to be updated.

**Comment by acidghost** [comment]
Hello there! I'm interested in reopening this PR and contributing with a patch to [SPARK-4362] based on this PR with the needed changes. I read [the wiki](https://cwiki.apache.org/confluence/display/SPARK/Contributing+to+Spark), should I simply follow those steps and create a new PR?

**Comment by acidghost** [comment]
@jkbradley  @srowen Do you think that the return type of `predictProbabilities` should be `scala.collection.Map[Double, Double]` or `Vector[Double]`?

**Comment by srowen** [comment]
I think the original Map was OK, IMHO.

**Comment by acidghost** [comment]
@srowen I just created PR #6761

**Pull request closed without merging**